### ISSUE TITLE
Checking for the absence of \ before a replacement isn't robust

### ISSUE THREAD

**Issue by DaGenix** [Zac-HD/js-regex#6]
The character class `\w` is replaced with `[a-zA-Z]` to restrict matches to ASCII. The code that does that replacement is: https://github.com/Zac-HD/js-regex/blob/81478d2480e1ac99a4dc6c311d2000b083afa6db/src/js_regex/_impl.py#L64

That code will replace any `\w` it finds with `[a-zA-Z]`, unless the `\w` is preceded by another backslash - as in `\\w` - in which case it won't do the replacement as this is equivalent to a literal backslash followed by a literal "w". However, this code fails to do the replacement for `\\\w` which is a literal backslash followed by the `\w` character class and should be translated into `\\[a-zA-Z]`.

Example:
```
>>> compile(r'\w').search('\u1234')
>>> compile(r'\\\w').search('\\\u1234')
<_sre.SRE_Match object; span=(0, 2), match='\\ሴ'>
```

**Comment by Zac-HD** [comment]
Yep... this is similar to #4 in that the only solution is to actually parse the regex and operate on that, rather than the text.  

It's definitely possible, but not likely to happen given that the jsonschema standard doesn't appear to really care about regex compatibility - I'd rather build a linter for incompatible syntax than a (much more difficult) translator.
